### ISSUE TITLE
Optional parameters are not immutable to being passed 'None'

### ISSUE THREAD

**Issue by MHHukiewitz** [aleph-im/aleph-client#84]
Take the following function signature as an example:
```python
create_post(
    account: Account,
    post_content,
    post_type: str,
    ref: Optional[str] = None,
    address: Optional[str] = settings.ADDRESS_TO_USE,
    channel: str = settings.DEFAULT_CHANNEL,
    session: Optional[ClientSession] = None,
    api_server: str = settings.API_HOST,
    inline: bool = True,
    storage_engine: StorageEnum = StorageEnum.storage,
) -> PostMessage:
```
The problem lies primarily with `address` & `channel`: One of them is not `Optional[str]`, where both provide default values. Another problem is that if I build another function on top like:
```python
async def post_or_amend_object(obj: T, account=None, channel=None):  # note, account and channel would be 'Optional'
    resp = await client.create_post(account, obj.content, channel=channel, ref=obj.item_hash)
```
Then the default values are being overwritten by `None`, so I had to to the following:
```python
    if account is None:
        account = FALLBACK_ACCOUNT
    if channel is None:
        channel = TEST_CHANNEL
    resp = await client.create_post(account, obj.content, channel=channel, ref=obj.item_hash)
```
which doesn't seem to serve the purpose of an `Optional` parameter. The same would be true for the `api_server` and `storage_engine` parameters. This does not seem to be an issue with the `get_posts()` and other `get_` methods. But the same is true for `create_program()` and `create_aggregate()` functions. IMHO, all of the functions of aleph-client should behave similarly when confronted with optional parameters, but because of this issue, this is not the case.

**Comment by hoh** [comment]
This is indeed an issue and we should try to fix it for the next release.

**Comment by MHHukiewitz** [comment]
Fixed in #132
